File: android2_lesson04/src/java/java/com/example/ama/android2_lesson04/ui/viewer/PVStartedServiceFragment.java
package com.example.ama.android2_lesson04.ui.viewer;

import android.content.Intent;
import android.os.Build;

import com.example.ama.android2_lesson04.ServiceTestApp;
import com.example.ama.android2_lesson04.background.BackgroundConstants;
import com.example.ama.android2_lesson04.background.service.LoadPictureCombinedService;
import com.example.ama.android2_lesson04.ui.viewer.base.BaseReceiverFragment;
import com.example.ama.android2_lesson04.utils.ResourcesUtils;

public class PVStartedServiceFragment extends BaseReceiverFragment {

    public static PVStartedServiceFragment newInstance() {
        return new PVStartedServiceFragment();
    }

    @Override
    protected void onLoadImagesClick() {
        startMyService();
    }

    private void startMyService() {
        if (getActivity() != null) {
            Intent intent = new Intent(ServiceTestApp.getInstance(), LoadPictureCombinedService.class);
            intent.putExtra(BackgroundConstants.EXTRA_KEY, ResourcesUtils.getUrlsArray());
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                getActivity().startForegroundService(intent);
            } else {
                getActivity().startService(intent);
            }
        }
    }
}

File: android2_lesson04/src/kotlin/java/com/example/ama/android2_lesson04/ui/viewer/PVStartedServiceFragment.kt
package com.example.ama.android2_lesson04.ui.viewer

import android.content.Intent
import com.example.ama.android2_lesson04.R
import com.example.ama.android2_lesson04.ServiceTestApp
import com.example.ama.android2_lesson04.background.utils.EXTRA_KEY
import com.example.ama.android2_lesson04.background.service.LoadPictureCombinedService
import com.example.ama.android2_lesson04.ui.viewer.base.BaseReceiverFragment

class PVStartedServiceFragment : BaseReceiverFragment() {

    companion object {
        fun newInstance() = PVStartedServiceFragment()
    }

    override fun onLoadImagesClick() {
        startService()
    }

    private fun startService() {
        val intent = Intent(ServiceTestApp.instance, LoadPictureCombinedService::class.java)
        intent.putExtra(EXTRA_KEY, resources.getStringArray(R.array.image_urls))
        activity?.startService(intent)
    }
}